llvm/docs/LangRef.rst

```diff
@@ -1741,6 +1741,20 @@ Currently, only the following parameter attributes are defined:
 
     This attribute cannot be applied to return values.
 
+``dead_on_return``
```

Comment:
The wording here should be a lot more similar to dead_on_unwind, describing what the precise operational semantics of the attribute are.

Reply:
Tried to improve the wording, hope it aligns more now. PR description updated, too (I got slightly threw off by the extra temporary allocation done in the callee in the unoptimized IR when targeting AArch64).